### PR TITLE
Reworked and centralized the document link generation.

### LINKED ISSUE #1823: Integration bumblebee SPV
Aus der SPV verlinkte Dokumente sollten auch im bumblebee overlay geöffnet werden:

Betroffen sind
- Anhänge (Antragsansicht/Sitzungsansicht)
- Protokolle
- Protokollauszüge (Antragsansicht/Sitzungsansicht)

### LINKED ISSUE #1945: Dokumentverlinkung bei Aufgaben zeigt nicht auf Dokumentvorschau.
bildschirmfoto_2016-06-17_um_12_40_24

### PR DESCRIPTION
**Pull request by phgross**
This PR reworks and centralize the document link generation, by using the `plone.app.contentlisting` ContentListing functionality. The following adjustments were necessary:
 - Extending the existing `OpengeverCatalogContentListingObject` Adapter with a new method `render_link`.
 - Adding a specific `ContentListingObject` Adapter for documentish objects (documents, mails)

And adjust all the places were a link documents gets created or rendered:
 - All tabbedview helpers, which were used in different document listings
 - Dossier overviews
 - Proposal and Submittedproposal overviews
 - Task overview
 - Meeting overview (protocols, manually generated excerpts, agenda_items attachments and excerpts).

This PR has the following positive consequences:
 - Bumblebee integration for tasks (fixes #1945)
 - Bumblebee integration for SPV (fixes #1823)
 - The document tooltip is now available on every document link.

@deiferni @lukasgraf

### CODE REVIEW

File: opengever/base/tests/test_contentlisting.py [hunk 1]
```diff
@@ -87,6 +87,93 @@ def assertCropping(self, size, value):
...
+        document_a = create(Builder('document'))
+        document_b = create(Builder('document').removed())
+        dossier = create(Builder('dossier'))
+
+        self.assertFalse(IContentListingObject(obj2brain(document_a)).is_removed)
+        self.assertTrue(
+            IContentListingObject(obj2brain(document_b, unrestricted=True)).is_removed)
+        self.assertFalse(IContentListingObject(obj2brain(dossier)).is_removed)
+
+    def test_get_breadcrumbs_returns_titles_joined_with_grater_than_sign(self):
```
Comment (deiferni) [inline opengever/base/tests/test_contentlisting.py]: grater -> greater

File: opengever/base/tests/test_contentlisting.py [hunk 2]
```diff
@@ -87,6 +87,93 @@ def assertCropping(self, size, value):
...
+            'Ordnungssystem > 1. Ablage 1 > hans m\xc3\xbcller > Anfrage Meier',
+            IContentListingObject(document).get_breadcrumbs())
+
+
+class TestBrainContentListingRenderLink(FunctionalTestCase):
+
+    def setUp(self):
+        super(TestBrainContentListingRenderLink, self).setUp()
+
+        # Because the DocumentLinkRenderer is already testet in a separate
```
Comment (deiferni) [inline opengever/base/tests/test_contentlisting.py]: tested

File: opengever/document/contentlisting.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    def is_trashed(self):
+        return ITrashed.providedBy(self._realobject)
+
+    @property
+    def is_removed(self):
+        removed_states = [Document.removed_state, OGMail.removed_state]
+        return api.content.get_state(self) in removed_states
+
+    def ContentTypeClass(self):
+        """Here we set the correct content type class so that documents with
```
Comment (deiferni) [inline opengever/document/contentlisting.py]: plz rephrase as a comman, e.g.:

Return the css class for the MIME type of the document's file.

File: opengever/document/tests/test_contentlisting.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+    def test_overlay_title_is_document_title_utf8_encoded(self):
+        document = create(Builder('document')
+                          .titled('Anfrage B\xc3\xbcchel'.decode('utf-8')))
+
+        self.assertEquals(
+            'Anfrage B\xc3\xbcchel',
+            IContentListingObject(document).get_overlay_title())
+
+    def test_get_breadcrumbs_returns_titled_joined_with_a_grather_than_symbol(self):
```
Comment (deiferni) [inline opengever/document/tests/test_contentlisting.py]: grather

File: opengever/document/tests/test_renderer.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        mail = create(Builder('mail'))
+
+        browser.open_html(DocumentLinkRenderer(document).render())
+        self.assertIn('PDF Preview', browser.css('.tooltip-links a').text)
+
+        browser.open_html(DocumentLinkRenderer(mail).render())
+        self.assertNotIn('PDF Preview', browser.css('.tooltip-links a').text)
+
+    @browsing
+    def test_preview_link_is_only_available_when_pdfconvertes_is_active(self, browser):
```
Comment (deiferni) [inline opengever/document/tests/test_renderer.py]: pdfconvertes

**Comment by deiferni** [comment]
👍

**Comment by phgross** [comment]
@deiferni updated ...

File: opengever/document/renderer.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+            return 'showroom-item'
+        return ''
+
+    def get_showroom_data(self):
+        return 'data-showroom-target="{overlay_url}" ' \
+            'data-showroom-title="{title}"'.format(
+                overlay_url=self.document.get_overlay_url(),
+                title=self.document.get_overlay_title())
+
+    def render(self):
```
Comment (lukasgraf) [inline opengever/document/renderer.py]: @phgross @deiferni 
Geez, use a page template!

Besides preventing the XSS attacks via several of those user-controlled parameters, that would also make the unicode error caused by `self.document.get_breadcrumbs()`'s easier to deal with.

**Pull request merged**